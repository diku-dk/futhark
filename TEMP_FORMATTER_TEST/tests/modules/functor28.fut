module type newreal = {
  type t
  val f32 : f32 -> t
}

module type newint = {
  type t
  val f32 : f32 -> t
}

module newf32 : newreal with t = f32 = {
  type t = f32
  def f32 = f32.f32
}

module newi32 : newint with t = i32 = {
  type t = i32
  def f32 = i32.f32
}

module type mixture = {
    module V : newreal
    module I : newint
}

module em (P: mixture) = {
  module mixture = P
}

module k_means_mixture (P: mixture) = {
    module V = P.V
    module I = P.I
}

module foo = {
  module V = newf32
  module I = newi32
}

module bar = k_means_mixture foo
module baz = bar : mixture
module k_means_em = em baz

def main (x: k_means_em.mixture.V.t) = x
