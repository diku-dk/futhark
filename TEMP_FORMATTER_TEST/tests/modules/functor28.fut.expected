84ba4e6a25920c4ddc60be647536173c
